### DISCUSSION TITLE
jplag between students commits

### DISCUSSION THREAD

**Discussion by rafaelcorsi** [jplag/JPlag#1974] [Q&A]
Hello,

Does anyone have a solution for analyzing student submissions made on Git that include a development history of the software? I just encountered a case where plagiarism occurred in a middle version of the program's development. Where one group continued to evolve the code to achieve a higher grade, while the other maintained the plagiarized code.

```
+---------------------+     +---------------------+
|      Group A        |     |      Group B        |
|                     |     +---------------------+
|                     |     | Initial Commit      |
|                     |     | (same initial code) |
|                     |     +---------------------+
|                     |     | Commit 1            |
|                     |     | (develop further)   |
+---------------------+     +---------------------+
| Initial Commit      | <---| Commit 2            |
| (plagiarized code)  |     | (develop further)   |
+---------------------+     +---------------------+
|                     |     | ...                 |
|                     |     +---------------------+
|                     |     | Final Commit        |
|                     |     | (new features)      |
+---------------------+     +---------------------+
```

**Comment by tsaglam** [comment]
One solution would be to check out and copy different versions of each submission with a script. Then, put all versions of each submission in a common folder.

```
rootA
|_ groupA-version1
|_ groupA-version2
rootB
|_ groupB-version1
|_ groupB-version2
```

Now you can use the `--old` argument to ensure that versions of the same submission are not checked with each other.

`--old rootA --old rootB`

(Note that you may need a non-old root directory with at least 2 submissions; this is currently a bug.)
